### ISSUE TITLE
Importing default module fails when using ES6 syntax

### ISSUE THREAD

**Issue by astoilkov** [Microsoft/TypeScript#3337]
I have tried importing `fs` module using `import fs from 'fs'` but this gives an error: `Module 'path' has no default export.'.

`import fs = require('fs');` works but this is not ES6 syntax. I would like to stick with ES6 syntax because it is getting more and more popular and therefore easier to understand.

Are there any plans for supporting ES6 modules that import the entire TypeScript module?

**Comment by ivogabe** [comment]
You should use
```typescript
import * as fs from 'fs';
```

You can use your syntax if the module has a default export. Example:
```typescript
// foo.ts
export default class Foo {
}

// bar.ts
import Foo from './foo';
new Foo();
```

**Comment by astoilkov** [comment]
Thanks! Closing this.

**Comment by tarruda** [comment]
I suggest making the existing `export =` syntax compatible with `export default`. Right now it doesn't seem possible to perform a default import of modules declared with the old syntax. 

Take the `tape` module for example. I can't import it using es6 syntax and the [existing](https://github.com/borisyankov/DefinitelyTyped/blob/master/tape/tape.d.ts#L9) definition file. The following directive:

```typescript
import tape from 'tape'
```

Fails with `TS1192: External module ''tape'' has no default export.`. If I change the definition file to use `export default tape` instead of `export = tape` then it works, but it will make the definition incompatible with typescript compilers < 1.5

**Comment by astoilkov** [comment]
+1

**Comment by danquirk** [comment]
We talked a lot about whether we can (or should) make export= and export default the same thing. You can see the discussion and conclusions here: https://github.com/Microsoft/TypeScript/issues/2242#issuecomment-83694181

**Comment by c9s** [comment]
I used 

    import * as assign from "object-assign";

And the error message becomes:

    error TS2497: Module ''object-assign'' resolves to a non-module entity and cannot be imported using this construct.

**Comment by c9s** [comment]
Setting "target" to "es5" and 

```
import fs = require('fs');
// or this
import assign = require('object-assign');
```

will work just fine, however this syntax will cause error with target = es6

**Comment by QuantumInformation** [comment]
What is wrong with:

`import {Game} from "./Game";`

game.ts:

`export class Game {`

I get the error:

`error TS1192: Module '"lib/Game"' has no default export.`

**Comment by mhegazy** [comment]
`game.ts` does not have a `default` export. if it did, it would have looked like:

```ts
export default class Game {
```

**Comment by QuantumInformation** [comment]
@mhegazy I am not wanting to use a default export here.

**Comment by kitsonk** [comment]
Is it `Game.ts` or `game.ts`... While some file systems allow case insensitivity in filenames, it can cause all sorts of strange issues in certain situations.

If it is `Game.ts` and the code is as you say it is, then it is unlikely that the `error TS1192` actually refers to the code you are providing, as it should work and not generate the error your are encountering.  It is likely something else is going on.

**Comment by mhegazy** [comment]
can you share your code then. the error is only shown if you have a default import.

**Comment by QuantumInformation** [comment]
@mhegazy sure its https://github.com/QuantumInformation/HTML5SpaceInvaders/blob/master/lib/Game.ts#L17

**Comment by QuantumInformation** [comment]
@mhegazy sorry please bear with me, I need to upload my local code after fixing smth.

**Comment by QuantumInformation** [comment]
@mhegazy switching to tsc vs ts-loader+webpack has fixed it for me.
https://github.com/QuantumInformation/HTML5SpaceInvaders/blob/master/Main.js#L3

**Comment by PendletonJones** [comment]
--allowSyntheticDefaultImports solved this for me, #2242 #5577
